app/models/custom_button.rb

```diff
@@ -12,7 +12,7 @@ class CustomButton < ApplicationRecord
   serialize :visibility
 
   validates :applies_to_class, :presence => true
-  validates :name, :description, :uniqueness => {:scope => [:applies_to_class, :applies_to_id]}, :presence => true
+  validates :name, :description, :uniqueness => {:scope => [:applies_to_class, :applies_to_id]}, :presence => true, if: :changed?
```

Comment:
as you now know, not a fan of the generic changed

not sure what is the best. here is something with my thoughts, not saying you need to do exactly this
```suggestion
  validates :name, :uniqueness => {:scope => [:applies_to_class, :applies_to_id]}, :if => :name_changed?
  validates :description, :uniqueness => {:scope => [:applies_to_class, :applies_to_id]}, :if => :description_changed?
  validates :name, :description, :presence => true
```